roofit/histfactory/src/HistFactoryModelUtils.cxx

```diff
@@ -185,11 +182,8 @@ namespace HistFactory{
     bool verbose=false;
 
     // Find the servers of this channel
-    //TIterator* iter = channel->serverIterator();
-    TIterator* iter = channel->getComponents()->createIterator(); //serverIterator();
     bool FoundParamHistFunc=false;
-    RooAbsArg* paramfunc_arg = NULL;
-    while(( paramfunc_arg = (RooAbsArg*) iter->Next() )) {
+    for (auto const *paramfunc_arg : static_range_cast<RooAbsArg *>(*channel->getComponents())) {
```

Comment:
```suggestion
    for (auto const *paramfunc_arg : *channel->getComponents()) {
```
RooAbsArg is already the class of the components (RooAbsArg is the base class of the other RooFit object classes), so this cast is unnecessary.